libraries/tools/kotlin-gradle-plugin/src/test/kotlin/org/jetbrains/kotlin/gradle/fixtures/FakeGradleProvider.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.fixtures
+
+import org.gradle.api.Transformer
+import org.gradle.api.provider.Provider
+import java.util.function.BiFunction
+
+class FakeGradleProvider<T>(private val v: (()-> T)?): Provider<T> {
```

Comment:
Maintaining this may be painful, I'd rather use Gradle `ProjectBuilder` to create `Project` instance and then to create providers. WDYT?